stl/inc/chrono

```diff
@@ -2209,6 +2221,2957 @@ namespace chrono {
+    _NODISCARD constexpr bool operator>=(const sys_time<_Duration>& _Left, const leap_second& _Right) noexcept {
+        return !(_Left < _Right.date());
+    }
+
+#ifdef __cpp_lib_concepts
+    // clang-format off
+    template <class _Duration>
+        requires three_way_comparable_with<sys_seconds, sys_time<_Duration>>
+    _NODISCARD constexpr auto operator<=>(
+        const leap_second& _Left, const sys_time<_Duration>& _Right) noexcept {
+        // clang-format on
+        return _Left.date() <=> _Right;
+    }
+    _NODISCARD constexpr strong_ordering operator<=>(const leap_second& _Left, const leap_second& _Right) noexcept {
+        return _Left.date() <=> _Right.date();
+    }
+#endif // __cpp_lib_concepts
+
+    // [time.zone.link]
+
+    // CLASS time_zone_link
+    class time_zone_link {
+    public:
+        explicit time_zone_link(string_view _Name_, string_view _Target_) : _Name(_Name_), _Target(_Target_) {}
+
+        time_zone_link(time_zone_link&&) = default;
+        time_zone_link& operator=(time_zone_link&&) = default;
+
+        _NODISCARD string_view name() const noexcept {
+            return _Name;
+        }
+
+        _NODISCARD string_view target() const noexcept {
+            return _Target;
+        }
+
+    private:
+        string _Name;
+        string _Target;
+    };
+
+    _NODISCARD inline bool operator==(const time_zone_link& _Left, const time_zone_link& _Right) noexcept {
+        return _Left.name() == _Right.name();
+    }
+
+#ifdef __cpp_lib_concepts
+    _NODISCARD inline strong_ordering operator<=>(const time_zone_link& _Left, const time_zone_link& _Right) noexcept {
+        return _Left.name() <=> _Right.name();
+    }
+#endif // __cpp_lib_concepts
+
+    // [time.zone.db]
+
+    _NODISCARD inline string _Tzdb_generate_current_zone() {
+        unique_ptr<__std_tzdb_current_zone_info, _Tzdb_deleter<__std_tzdb_current_zone_info>> _Info{
+            __std_tzdb_get_current_zone()};
+        if (_Info == nullptr) {
+            _Xbad_alloc();
+        } else if (_Info->_Err == __std_tzdb_error::_Win_error) {
+            _XGetLastError();
+        } else if (_Info->_Err == __std_tzdb_error::_Icu_error) {
+            _Xruntime_error("Internal error loading IANA database information");
+        }
+
+        return {_Info->_Tz_name};
+    }
+
+    template <class _Ty>
+    _NODISCARD const _Ty* _Locate_zone_impl(const vector<_Ty>& _Vec, string_view _Name) {
+        const auto _Result = _STD find_if(_Vec.begin(), _Vec.end(), [&](auto& _Tz) { return _Tz.name() == _Name; });
+        return _Result == _Vec.end() ? nullptr : &*_Result;
+    }
+
+    // STRUCT tzdb
+    struct tzdb {
+        string version;
+        vector<time_zone> zones;
+        vector<time_zone_link> links;
+        vector<leap_second> leap_seconds;
+        bool _All_ls_positive;
+
+        _NODISCARD const time_zone* locate_zone(string_view _Tz_name) const {
+            auto _Tz = _Locate_zone_impl(zones, _Tz_name);

```

Comment:
Is there a specific reason not to use if with initializer

Reply:
No specific reason - I think it's well-supported by all of our C++20 compilers. (I am cautious about using new compiler features "unnecessarily", but this one hasn't been problematic AFAIK.) At this point, it's just time pressure - we need to land this *soon*. We can consider cleanup PRs in the future.

Reply:
I generally see variable declarations/initializations in `if` tests less frequently than splitting it out into a separate variable in the code base, but I'm sure there are occurrences of both.  I went with my preference (splitting out the variable declaration/initialization) but have no objections if there are strong feelings otherwise.

Reply:
No objections, I usually find them less usefull because often one needs the variable later on. 

This was one of the rare cases where it would have worked.